In [3]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
%matplotlib inline
from sklearn.ensemble import RandomForestClassifier
from sklearn.model_selection import train_test_split
from sklearn.tree import DecisionTreeClassifier
from sklearn.metrics import classification_report, confusion_matrix

In [5]:
warBat = pd.read_csv('/Users/anthonyzippay/Desktop/warBatNA.csv')

In [6]:
warBat.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 49719 entries, 0 to 49718
Data columns (total 51 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   Unnamed: 0          49719 non-null  int64  
 1   X                   49719 non-null  int64  
 2   name_common         49719 non-null  object 
 3   age                 49719 non-null  int64  
 4   mlb_ID              49719 non-null  int64  
 5   playerID            49719 non-null  object 
 6   year_ID             49719 non-null  int64  
 7   team_ID             49719 non-null  object 
 8   stint_ID            49719 non-null  int64  
 9   lg_ID               49719 non-null  object 
 10  PA                  49550 non-null  float64
 11  G                   49719 non-null  int64  
 12  Inn                 49719 non-null  float64
 13  runs_bat            49719 non-null  float64
 14  runs_br             49719 non-null  float64
 15  runs_dp             49719 non-null  float64
 16  runs

In [10]:
warBat1 = warBat['year_ID', 'Salary']

KeyError: ('year_ID', 'Salary')

In [7]:
x = warBat.drop('year_ID', axis=1)
y = warBat['year_ID']

In [8]:
x_train, x_test, y_train, y_test = train_test_split(x,y, test_size=0.3, random_state=76)

In [9]:
forest = RandomForestClassifier(n_estimators=500, random_state=76)
forest.fit(x_train, y_train)

ValueError: could not convert string to float: 'Ruben Rivera'

In [20]:
forestPredictions = forest.predict(x_test)
print(confusion_matrix(y_test, forestPredictions))
print(classification_report(y_test, forestPredictions))

[[  0   0   1   0   3   3   3   2   2   0   0   0   0   0   0   0   0   0
    0   0   0   0   0   0   0   0   0   0   0]
 [  0   0   1   4   8  12  12   8   5   0   1   0   0   0   0   0   0   1
    0   0   0   0   0   0   0   0   0   0   0]
 [  0   0   1   8  38  41  55  26   8  10   7   1   1   3   0   0   0   1
    0   0   0   0   0   0   0   0   0   0   0]
 [  0   0   6  18  74 105 105  74  44  22  17   6   5   5   1   0   0   1
    0   0   0   0   0   0   0   0   0   0   0]
 [  0   2   7  29 112 209 199 138  82  35  28  12   9   5   4   2   2   1
    0   0   0   0   0   0   0   0   0   0   0]
 [  0   2   8  24 100 260 316 220 154  64  36  17  17   6   3   2   0   1
    1   1   0   0   0   0   0   0   0   0   0]
 [  0   3   6  28 106 236 326 330 221 109  66  25  27  14   7   2   2   3
    0   1   0   0   0   1   0   0   0   1   0]
 [  0   3  10  37  76 194 311 307 278 142  89  42  25  24  11   8   7   1
    2   0   0   0   0   0   0   1   0   0   0]
 [  0   1   7  15  49 149 239 26

/opt/anaconda3/lib/python3.7/site-packages/sklearn/metrics/_classification.py:1272: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))


100% accuracy with this model. year_ID was our x. What does that mean?

using age as our x, we found that it was only 14% accurate. the random forest wasnt any better. 

when I try doing this with Salary I get an error. "ValueError: Unknown label type: 'continuous". I havent figured out how to deal with it yet.